toolchain/check/testdata/function/builtin/fail_redefined.carbon

```diff
@@ -43,18 +43,18 @@ fn C(n: i32, m: i32) -> i32 = "int.sadd";
 // CHECK:STDOUT: constants {
 // CHECK:STDOUT:   %int_32: Core.IntLiteral = int_value 32 [concrete]
 // CHECK:STDOUT:   %i32: type = class_type @Int, @Int(%int_32) [concrete]
-// CHECK:STDOUT:   %A.type: type = fn_type @A [concrete]
-// CHECK:STDOUT:   %A: %A.type = struct_value () [concrete]
-// CHECK:STDOUT:   %.type.b6a92a.1: type = fn_type @.1 [concrete]
-// CHECK:STDOUT:   %.d852be.1: %.type.b6a92a.1 = struct_value () [concrete]
-// CHECK:STDOUT:   %B.type: type = fn_type @B [concrete]
-// CHECK:STDOUT:   %B: %B.type = struct_value () [concrete]
-// CHECK:STDOUT:   %.type.b6a92a.2: type = fn_type @.2 [concrete]
-// CHECK:STDOUT:   %.d852be.2: %.type.b6a92a.2 = struct_value () [concrete]
-// CHECK:STDOUT:   %C.type: type = fn_type @C [concrete]
-// CHECK:STDOUT:   %C: %C.type = struct_value () [concrete]
-// CHECK:STDOUT:   %.type.b6a92a.3: type = fn_type @.3 [concrete]
-// CHECK:STDOUT:   %.d852be.3: %.type.b6a92a.3 = struct_value () [concrete]
+// CHECK:STDOUT:   %A.type.00d7e7.1: type = fn_type @A.1 [concrete]
```

Comment:
It would have been nice to split the decl name change and all the test changes for it into a separate PR, but no need now. Hopefully I reviewed all the appropriate test changes for `destroy`, as there's a lot to skim through.